packages/amplify-graphql-schema-generator/src/__tests__/schema-overrides.test.ts

```diff
@@ -397,6 +398,308 @@ describe('apply schema overrides for models with refersTo', () => {
+    const editedSchema = `
+              type MyPost @refersTo(name: "Post") @model {
+                  id: ID!
+                  name: String @refersTo(name: "nameField")
+              }
+          `;
+    const editedDocument = parse(editedSchema);
+    const updatedDocument = applySchemaOverrides(document, editedDocument);
+    stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
+  });
+
+  it('should retain renaming of fields referenced with relational directive', () => {
+    const document = parse(`
+              type Profile @model {
+                  id: String! @primaryKey
+                  details: String
+                  userId: String @index(sortKeyFields: ["id"])
+              }
+          
+              type User @refersTo(name: "Users") @model {
+                  id: String! @primaryKey
+                  name: String @refersTo(name: "nameField")
+              }
+        `);
+    const editedSchema = `type Portfolio @refersTo(name: "Profile") @model {
+            myId: String! @refersTo(name: "id") @primaryKey
+            details: String
+            myUserId: String @refersTo(name: "userId") @index(sortKeyFields: ["id"])
+            user: User @belongsTo(references: "myUserId")
+        }
+
+        type User @refersTo(name: "Users") @model {
+            id: String! @primaryKey
+            myName: String @refersTo(name: "nameField")
+            portfolio: Portfolio @hasOne(references: "myUserId")
+        }
+      `;
+    const editedDocument = parse(editedSchema);
+    const updatedDocument = applySchemaOverrides(document, editedDocument);
+    stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
+  });
+
+  it('should retain removing the renaming of fields referenced with relational directive', () => {
+    const document = parse(`
+              type Profile @model {
+                  myId: String! @refersTo(name: "id") @primaryKey
+                  details: String
+                  myUserId: String @refersTo(name: "userId") @index(sortKeyFields: ["id"])
+              }
+          
+              type User @refersTo(name: "Users") @model {
+                  id: String! @primaryKey
+                  myName: String @refersTo(name: "name")
+              }
+        `);
+    const editedSchema = `type Profile @model {
+              id: String! @primaryKey
+              details: String
+              userId: String @index(sortKeyFields: ["id"])
+              user: User @belongsTo(references: "userId")
+          }
+      
+          type User @refersTo(name: "Users") @model {
+              id: String! @primaryKey
+              name: String
+              profile: Profile @hasOne(references: "userId")
+        }`;
+    const editedDocument = parse(editedSchema);
+    const updatedDocument = applySchemaOverrides(document, editedDocument);
+    stringsMatchWithoutWhitespace(print(updatedDocument), editedSchema);
+  });
+
+  it('should retain relational directives', () => {
```

Comment:
Please rename this test -- you're asserting the error condition, and it wasn't until I read the assertion at the bottom that I realized you weren't actually going to allow a refersTo on a relational field :)